workflows/taxprofiler.nf

```diff
@@ -111,67 +111,67 @@ workflow TAXPROFILER {
     ch_multiqc_files = Channel.empty()
 
     // Validate input files and create separate channels for FASTQ, FASTA, and Nanopore data
-    samplesheet
-        .branch { meta, run_accession, instrument_platform, fastq_1, fastq_2, fasta ->
-            //println "Mapping: meta=$meta, run_accession=$run_accession, instrument_platform=$instrument_platform, fastq_1=$fastq_1, fastq_2=$fastq_2, fasta=$fasta"
-
-        meta.run_accession = run_accession
-        meta.instrument_platform = instrument_platform
+    ch_input = samplesheet
+        .map { meta, run_accession, instrument_platform, fastq_1, fastq_2, fasta ->
+            meta.run_accession = run_accession
+            meta.instrument_platform = instrument_platform
 
-        // Define single_end based on the conditions
-        meta.single_end = (fastq_1 && !fastq_2 && instrument_platform != 'OXFORD_NANOPORE')
+            // Define single_end based on the conditions
+            meta.single_end = ( fastq_1 && !fastq_2 && instrument_platform != 'OXFORD_NANOPORE' )
 
-        // Define is_fasta based on the presence of fasta
-        meta.is_fasta = fasta ? true : false
+            // Define is_fasta based on the presence of fasta
+            meta.is_fasta = fasta ? true : false
 
-        if (!meta.is_fasta && !fastq_1) {
-            error("ERROR: Please check input samplesheet: entry `fastq_1` doesn't exist!")
-        }
-        if (meta.instrument_platform == 'OXFORD_NANOPORE' && fastq_2) {
-            error("Error: Please check input samplesheet: for Oxford Nanopore reads entry `fastq_2` should be empty!")
+            if ( !meta.is_fasta && !fastq_1 ) {
+                error("ERROR: Please check input samplesheet: entry `fastq_1` doesn't exist!")
+            }
+            if ( meta.instrument_platform == 'OXFORD_NANOPORE' && fastq_2 ) {
+                error("Error: Please check input samplesheet: for Oxford Nanopore reads entry `fastq_2` should be empty!")
+            }
+            if ( meta.single_end && fastq_2 ) {
+                error("Error: Please check input samplesheet: for single-end reads entry `fastq_2` should be empty")
+            }
+            return [ meta, run_accession, instrument_platform, fastq_1, fastq_2, fasta ]
         }
-        if (meta.single_end && fastq_2) {
-            error("Error: Please check input samplesheet: for single-end reads entry `fastq_2` should be empty")
+        .branch { meta, run_accession, instrument_platform, fastq_1, fastq_2, fasta ->
+            fastq_se: meta.single_end
```

Comment:
Any reason why to split fastq_pe and fastq_se when you immediately merge after?

Reply:
Good question. I think I just followed the formal input_check logic without thinking. You are right, I will modify this part.